dex/src/state.rs

```diff
@@ -2686,8 +2680,168 @@ impl State {
+            pc_vault,
+            spl_token_program,
+            fee_tier,
+            vault_signer,
+        } = args;
+
+        let order_id = req_q.gen_order_id(instruction.limit_price.get(), instruction.side);
+        let native_pc_qty_locked = match instruction.side {
+            Side::Bid => {
+                let lock_qty_native = instruction.max_native_pc_qty_including_fees;
+                Some(lock_qty_native)
+            }
+            Side::Ask => None,
+        };
+
+        let request = RequestView::NewOrder {
+            side: instruction.side,
+            order_type: OrderType::ImmediateOrCancel,
+            order_id,
+            fee_tier,
+            self_trade_behavior: SelfTradeBehavior::AbortTransaction,
+            // Pass dummy account - system program pubkey
+            owner: system_program::ID.to_aligned_bytes(),
+            // Pass 0
+            owner_slot: 0,
+            max_coin_qty: instruction.max_coin_qty,
+            native_pc_qty_locked,
+            client_order_id: None,
+        };
+
+        let mut limit = instruction.limit;
+        let mut proceeds = RequestProceeds::zero();
+        let _unfilled_portion = order_book_state.process_orderbook_request(
+            &request,
+            &mut event_q,
+            &mut proceeds,
+            &mut limit,
+        )?;
+
+        let market_state = order_book_state.market_state;
+
+        let coin_lot_size = market_state.coin_lot_size;
+
+        let RequestProceeds {
+            coin_unlocked: _,
+            coin_credit,
+
+            native_pc_unlocked: _,
+            native_pc_credit,
+
+            coin_debit,
+            native_pc_debit,
+        } = proceeds;
+
+        let abort = match instruction.side {
+            Side::Bid if coin_credit < instruction.min_coin_qty => true,
+            Side::Ask if native_pc_credit < instruction.min_native_pc_qty => true,
+            _ => false,
+        };
+
+        if abort {
+            solana_program::msg!("Min amount requested not met! Aborting");
+            return Err(DexErrorCode::MinAmountNotMet.into());
+        };
+
+        // Amount that user deposits into the program
+        let deposit_amount;
+        // Amount that user receives after the exchange 
+        let withdraw_amount;
+
+        // Token accounts for transfers
+        let deposit_vault;
+        let withdraw_vault;
+        let deposit_source;
+        let withdraw_destination;
+
+        match instruction.side {
+            Side::Bid => {
+                deposit_amount = native_pc_debit;
+                withdraw_amount = coin_credit.checked_mul(coin_lot_size).unwrap();
+                deposit_vault = pc_vault.token_account();
+                deposit_source = pc_wallet.token_account();
+                withdraw_destination = coin_wallet.token_account();
+                withdraw_vault = coin_vault.token_account();
+
+                market_state.pc_deposits_total += deposit_amount;
```

Comment:
@sintemal this is where it gets updated. @a-creation  pointed out to me that there's potentially a mismatch here as a result of fees

Reply:
Yes exactly, this is what was irritating me. I think we should just substract the fees in the `matching.rs` as before.

Reply:
@sintemal I'm leaning towards that too. Do you think it's fine if pc_deposits_total is temporarily negative/underflowed then?

Reply:
Yes, I think this should be fine